app/controller/toolbar/CircleSelectionTool.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * This class is the controller for the CircleSelectionController.
+ */
+Ext.define('CpsiMapview.controller.toolbar.CircleSelectionController', {
+    extend: 'Ext.app.ViewController',
+
+    alias: 'controller.cmv_circle_selection_toolbar',
+
+    /**
+     * Handles the click on the `Apply` button by firing
+     * the `circleSelectApply` event of the button that created
+     * the toolbar
+     */
+    handleApply: function () {
+        var me = this;
+        var feat = me.getView().feature;
+        me.getView().fireEvent('circleSelectApply', feat);
```

Comment:
Why is this `fireEvent` delegated to the view instance?

Reply:
I had some problems listening to the event, which was based on my little experience with Ext. But was now solved in https://github.com/compassinformatics/cpsi-mapview/pull/177/commits/b79cde00e7b4d84cc737d59866c7570cad768ee5